src/api/gists.rs

```diff
@@ -122,6 +123,79 @@ impl<'octo> GistsHandler<'octo> {
     pub fn list_commits(&self, gist_id: impl Into<String>) -> list_commits::ListCommitsBuilder {
         list_commits::ListCommitsBuilder::new(self, gist_id.into())
     }
+
+    /// Check if the given is gist is already starred by the authenticated user.
+    /// See [GitHub API Documentation][docs] more information about response
+    /// data.
+    ///
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// let is_starred: bool = octocrab::instance()
+    ///     .gists()
+    ///     .is_starred("00000000000000000000000000000000")
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// [docs]: https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#check-if-a-gist-is-starred
+    pub async fn is_starred(&self, gist_id: impl AsRef<str>) -> Result<bool> {
+        let gist_id = gist_id.as_ref();
+        let response = self
+            .crab
+            ._get(format!("/gists/{gist_id}/star"))
+            .await?;
+        // Gist API returns 204 (NO CONTENT) if a gist is starred
+        Ok(response.status() == StatusCode::NO_CONTENT)
```

Comment:
Do we need a `StarGistHandler` to wrap the http specifics?

Reply:
I'm not sure what you mean, what other semantics?

Reply:
Ah, I meant I wasn't sure if you preferred hiding the http status check behind a builder. I'm not sure what the "correct" approach was so I went with the simplest one.

Reply:
No, a builder should only be used if there's optional parameters, if there's only required parameters (or no parameters), then a builder is not needed.